### PR TITLE
Pin python dependencies with Poetry

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
Ported from https://github.com/canonical/mysql-operator/pull/192

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,69 @@
...
+[tool.poetry.dependencies]
+python = "^3.10"
+ops = "^2.5.0"
+lightkube = "^0.14.0"
+tenacity = "^8.2.2"
+boto3 = "^1.28.22"
+
+[tool.poetry.group.charm-libs.dependencies]
+# data_platform_libs/v0/data_interfaces.py
+ops = ">=2.0.0"
```
Comment (paulomach) [inline pyproject.toml:20]: question: how does setting `ops">=2.0.0"` here interact with `ops = "^2.5.0"` on line 13?

**Review by paulomach — APPROVED** [review]
nice. just a doubt on poetry

**Review by shayancanonical — APPROVED** [review]
looks great!

**Review by taurus-forever — APPROVED** [review]
Nice!

**Pull request merged**